recipes-core/initscripts/initscripts-1.0/hostname.sh

```diff
@@ -5,13 +5,10 @@
 # Version:      @(#)hostname.sh  1.10  26-Feb-2001  miquels@cistron.nl
 #
 
-if [ -f /etc/hostname ] && [ -r /etc/hostname ];then
-    HOSTNAME=$(cat /etc/hostname)
-else
-    HOSTNAME=
-fi
-
+# Check if there is a hostname in ni-rt.ini
+HOSTNAME=$(/usr/local/natinst/bin/nirtcfg --get section=SystemSettings,token=Host_Name)
 if [ $? -eq 0 ] && [ -n "$HOSTNAME" ]; then
+    # Set the hostname based on ni-rt.ini
     hostname "$HOSTNAME"
```

Comment:
When we set the hostname here, does it automatically update the value of `/etc/hostname`? We need to ensure that the value in that file is kept in-sync with `ni-rt.ini`.

Reply:
Checked on a target, the /etc/hostname file does not exist, even after updating the hostname. I can update this script to create the file.